narwhals/dataframe.py

```diff
@@ -132,6 +133,7 @@ def join(
     ) -> Self:
         if how != "inner":
             raise NotImplementedError("Only inner joins are supported for now")
+        validate_same_library(self, other)
```

Comment:
rather than making two paths inside `validate_same_library`, could we just call this one with
```python
validate_same_library([self, other])
```
?

Reply:
Didn't think about it! 🤓